holidays/countries/tanzania.py

```diff
@@ -0,0 +1,301 @@
+    # John Pombe Magufuli Inauguration Day.
+    john_magufuli_inauguration = tr("Sikukuu ya Kuapishwa kwa John Pombe Magufuli")
+
+    # Tanzania General Election Day.
+    tz_election_day = tr("Sikukuu ya Uchaguzi Mkuu wa Tanzania")
+
+    # National Population and Housing Census Day.
+    phc_census_day = tr("Siku ya Sensa ya Kitaifa ya Watu na Makazi")
+
+    # John Pombe Magufuli's Funeral.
+    john_magufuli_funeral = tr("Mazishi cha John Pombe Magufuli")
+
+    special_public_holidays = {
+        2002: (AUG, 25, phc_census_day),
+        2015: (NOV, 5, john_magufuli_inauguration),
+        2020: (OCT, 28, tz_election_day),
+        2021: (
+            (MAR, 22, john_magufuli_funeral),
+            (MAR, 25, john_magufuli_funeral),
+        ),
+        2022: (AUG, 23, phc_census_day),
+    }
+
+    def __init__(self, *args, **kwargs):
+        ChristianHolidays.__init__(self)
+        InternationalHolidays.__init__(self)
+        IslamicHolidays.__init__(self, calendar=TanzaniaIslamicCalendar())
+        super().__init__(*args, **kwargs)
+
+    def _populate_bank_holidays(self):
+        # Written Law (Miscellaneous Amendments) (No. 2) Act, 1994.
+        if self._year <= 1993:
+            return None
+
+        # Sikukuu ya Pasaka.
+        # Status: In-Use.
+        # Only observed by financial institutions.
+
+        # Easter Sunday.
+        self._add_easter_sunday(tr("Sikukuu ya Pasaka"))
+
+    def _populate_public_holidays(self):
+        # Written Law (Miscellaneous Amendments) (No. 2) Act, 1994.
+        if self._year <= 1993:
+            return None
+
+        # In-lieus ("Badala ya %s") are observed on Monday should it fall on the weekends.
+        # Abrogated in Public Holidays Ordinance  No. 28 of 1966.
+        # Reinstitued in Written Law (Miscellaneous Amendments) (No. 2) Act, 1994.
+        # Claimed to be abrogated again on an unspecified date.
+
+        # Fixed Date Holidays.
+
+        # Mwaka Mpya.
+        # Status: In-Use.
+        # Abrogated in Public Holidays Ordinance No. 28 of 1966.
+        # Reinstitued prior to the Written Law (Miscellaneous Amendments) (No. 1) Act, 1993.
+
+        # New Year's Day.
+        self._add_new_years_day(tr("Mwaka Mpya"))
+
+        # Mapinduzi ya Zanzibar.
+        # Status: In-Use.
+        # Commemorates the fall of the Sultanate of Zanzibar on Jan 12, 1964.
+
+        # Zanzibar Revolution Day.
+        self._add_holiday_jan_12(tr("Mapinduzi ya Zanzibar"))
+
+        # Siku ya kumbukumbu ya Rais wa Kwanza wa Serikali ya Mapinduzi Zanzibar
+        # Sheikh Abeid Amant Karume.
+        # Status: In-Use.
+        # Commemorates the death of Abeid Amani Karume on Apr 7, 1972.
+        # Assumed to start via Government Notices 79 of 1977.
+
+        self._add_holiday_apr_7(
+            # The Sheikh Abeid Amani Karume Day.
+            tr(
+                "Siku ya kumbukumbu ya Rais wa Kwanza wa Serikali ya Mapinduzi Zanzibar "
+                "Sheikh Abeid Amant Karume"
```

Comment:
```suggestion
                "Sheikh Abeid Amani Karume"
```

Reply:
Oof, missed - this is hopefully fixed now.